### ISSUE TITLE
Documentation: "Coding" section question

### ISSUE THREAD

**Issue by IanQS** [LazyVim/LazyVim#1316]
1) I'm going through the documentation to set up lazyvim (I'm using Rust and Python) and I'm a tad bit lost when it comes to [lazyvim/plugins/coding](https://www.lazyvim.org/plugins/coding). There are many things I can add, and some things are already dependencies on others. What is the right way to do things in situations like this?

For a concrete example, `LuaSnip` adds `friendly-snippets` as a dependency. Should I include a separate config file for `friendly-snippets` in cases like this? I assume not, as all the relevant sections are embedded, but I want to be sure. 

TL;DR should I just include all of the configs, even if some are already included as dependencies in others

2) I'm also curious about what might happen if I embedded it and included a separate file containing a config for `friendly-snippets`. I can see a situation like this happening if I were to include `friendly-snippets` as a dependency e.g. make it `lazy_load` in the `LuaSnip` config, but also in a separate file.

---

Thanks for taking the time! I searched around for some `LazyVim` configs here on github to try and find answers to my questions, but it seems like everyone structures things differently and I'm a little lost

**Comment by jyuan0** [comment]
1. The plugins listed on [plugins/coding](https://www.lazyvim.org/plugins/coding) are the plugins that are enabled by default. You don't have to do anything to set them up besides install the LazyVim starter template using the steps here: https://www.lazyvim.org/installation. The same is true for the plugins under [Editor](https://www.lazyvim.org/plugins/editor), [UI](https://www.lazyvim.org/plugins/ui), etc.
For the plugins under the "Extras" section, which are not enabled by  default, you can enable them by adding the line highlighted under "To use this, add it to your **lazy.nvim** imports:". E.g., for [Python](https://www.lazyvim.org/extras/lang/python), you just need to add `{ import = "lazyvim.plugins.extras.lang.python" },` to `config/lazy.lua`.
> For a concrete example, LuaSnip adds friendly-snippets as a dependency. Should I include a separate config file for friendly-snippets in cases like this? I assume not, as all the relevant sections are embedded, but I want to be sure.
>
> TL;DR should I just include all of the configs, even if some are already included as dependencies in others

Your assumption is correct, you don't have to include a separate config for a plugin unless you want to change or add to a plugin config. The plugin settings you see listed are just to show you how it's being configured by default - you don't have to have a separate config unless you want to change or add something.

2. If you created another `config` for `friendly-snippets`, it would override the `config` you see listed on  [plugins/coding](https://www.lazyvim.org/plugins/coding):
```
    config = function()
      require("luasnip.loaders.from_vscode").lazy_load()
    end,
```
This is because only one `config` is allowed. The [Plugins](https://www.lazyvim.org/configuration/plugins) page should clarify things. The [lazy.nvim](https://github.com/folke/lazy.nvim) docs will also be helpful because it's the package manager being used. And just to avoid any confusion, `lazy.nvim` is the package manager, and `LazyVim` is this repo or Neovim distribution.

----
> I searched around for some LazyVim configs here on github to try and find answers to my questions, but it seems like everyone structures things differently and I'm a little lost

There is a lot of freedom with how to do things. Here are a few setups that might help:
- https://github.com/folke/dot/tree/master/nvim
- https://github.com/dpetka2001/dotfiles/tree/main/dot_config/nvim
- https://github.com/appelgriebsch/Nv

**Comment by dpetka2001** [comment]
@abeldekat's configuration is also worth mentioning.
- https://github.com/abeldekat/nvim

He does awesome work.

**Comment by IanQS** [comment]
Yes! Will do. I'm a little outdated with the whole GitHub and discussions thing; I'll open up a follow up in the discussions section
